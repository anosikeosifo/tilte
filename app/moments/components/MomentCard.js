require('../styles/moment_card.scss');

import React, { Component, PropTypes } from 'react';
import classNames from 'classnames';
import { FEED_IMAGE_PLACEHOLDERS, USER_AVATAR_PLACEHOLDER } from '../constants';
import ActorCard from '../../users/components/ActorCard';
import LocationDetailsTrigger from '../../location/components/LocationDetailsTrigger';
import LikeTrigger from '../../core/components/LikeTrigger';
import CommentTrigger from '../../comments/components/CommentTrigger';
import MoreInteractionsTrigger from '../../core/components/MoreInteractionsTrigger';
import update from 'immutability-helper';
import { Link } from 'react-router';

class MomentCard extends Component {
  componentWillMount() {
    this.setState(Object.assign({}, this.state, {
      cardInteractionActive: false
    }));
  }

  handleMouseEnter() {
    this.setState(update(this.state, {
      cardInteractionActive: { $set: true }
    }));
  }

  handleMouseLeave() {
    this.setState(update(this.state, {
      cardInteractionActive: { $set: false }
    }));
  }

  handleClick() {

  }

  render() {
    const locationInfo = {
      latitude: '',
      longitude: '',
      actor: this.props.user,
    };

    const classes = classNames({
      'event__wrap': true,
      [this.props.id]: true,
      'hover__active': this.state.cardInteractionActive
    });

    const EventDetailsURL = `/events/${ this.props.id }`;

    return(
      <article className='component__momentcard'>
        <Link to={ EventDetailsURL }>
          <figure className={ classes }
            onMouseEnter={ this.handleMouseEnter.bind(this) }
            onMouseLeave={ this.handleMouseLeave.bind(this) }
            onClick={ this.handleClick.bind(this) }>

            <section className='event_media_object'>
              <img src={ FEED_IMAGE_PLACEHOLDERS[0] } alt='' className='feed_img'/>
            </section>
            <figcaption>
              <section className='item__main'>
                <span className='item__metadata'>
                  <span className='timestamp'>3m</span>
                  &middot;
                  <span className='nested location'>
                    <LocationDetailsTrigger metadata={ locationInfo } />
                  </span>
                </span>
                <div className='item__content'>
                  { this.props.desription || "hello! welcome to tilte, the place where we share the most fun experiences!" }
                </div>
              </section>

              <section className='item__interactions'>
                <span className='nested user'>
                  <ActorCard actor={ this.props.user }/>
                </span>
                <div className='action__group'>
                  <span className='nested like'>
                    <LikeTrigger eventId={ this.props.id } isEnabled={ !this.props.is_favorite } uiState={ this.props.is_favorite ? "active" : "default" } likeAction={ this.props.actions.like } actor={ this.props.currentUser } />
                  </span>
                  <span className='nested comment'>
                    <CommentTrigger />
                  </span>
                  <span className='nested  more'>
                    <MoreInteractionsTrigger />
                  </span>
                </div>
              </section>
            </figcaption>
          </figure>
        </Link>
      </article>
    );
  }
}

MomentCard.propTypes = {
  actions: PropTypes.object.isRequired,
}

export default MomentCard;
