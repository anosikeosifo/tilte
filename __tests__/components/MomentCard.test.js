import React from 'react';
import renderer from 'react-test-renderer';
import MomentCard from '../../app/components/MomentCard';
import { shallow } from 'enzyme';


describe('MomentCard Component', () => {
  const momentId = '10';
  const actionProps = { like: () => ({"test": "test"}) };
  const momentProps = { id: '10', user: { avatarUrl: null } }

  const momentCardTree = shallow(
    <MomentCard id={ momentId } actions={actionProps} {...momentProps}/>
  );
  console.log('momentCardTree: ', momentCardTree);
  // let momentCardTree = component.toJSON();

  // it('should matched expected initial state', () => {
  //   expect(momentCardTree).toMatchSnapshot();
  // });

  it('should open the moment detail view when clicked', () => {
    const linkURL = momentCardTree.find('a')
                      .props.href;
    const card = momentCardTree.find('.moment__wrap')
    console.log('card: ', card);
    
    card.props.onClick();
    expect(window.location.pathname).toBe.equalTo(linkURL);
  });
});