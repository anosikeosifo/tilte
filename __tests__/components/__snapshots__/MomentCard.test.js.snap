exports[`MomentCard Component should matched expected initial state 1`] = `
<article
  className="component__moment">
  <a
    onClick={[Function]}
    style={Object {}}>
    <figure
      className="10 moment__wrap"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}>
      <section
        className="moment_media_object">
        <img
          alt=""
          className="feed_img"
          src="http://localhost:9000/images/feed1.jpg" />
      </section>
      <figcaption>
        <section
          className="item__main">
          <span
            className="item__metadata">
            <span
              className="timestamp">
              3m
            </span>
            Â·
            <span
              className="nested location">
              <section
                className="component__location_details_trigger">
                Lagos, Nigeria
              </section>
            </span>
          </span>
          <div
            className="item__content">
            hello! welcome to tilte, the place where we share the most fun experiences!
          </div>
        </section>
        <section
          className="item__interactions">
          <span
            className="nested user">
            <section
              className="component__actor__card">
              <span
                className="user__image__wrap">
                <img
                  className="image"
                  src="http://localhost:9000/images/hamburg.jpg" />
              </span>
              <span
                className="user__name__wrap" />
            </section>
          </span>
          <div
            className="action__group">
            <span
              className="nested like">
              <section
                className="component__like__triger enabled"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}>
                <svg
                  className="icon"
                  height="24px"
                  style={
                    Object {
                      "display": "inline-block",
                      "verticalAlign": "middle",
                    }
                  }
                  viewBox="0 0 1024 1024"
                  width="24px">
                  <path
                    d={undefined}
                    style={
                      Object {
                        "fill": "#c1c1c1",
                      }
                    } />
                </svg>
              </section>
            </span>
            <span
              className="nested comment">
              <section
                className="component__comment__triger" />
            </span>
            <span
              className="nested  more">
              <section
                className="component__more__interactions" />
            </span>
          </div>
        </section>
      </figcaption>
    </figure>
  </a>
</article>
`;
